### PR TITLE
Update to NextJS 12

### LINKED ISSUE #2401: Update to nextjs 12
https://nextjs.org/blog/next-12 is out, let's update:

- Rust Compiler: ~3x faster Fast Refresh and ~5x faster builds
- Middleware (beta): Enabling full flexibility in Next.js with code over configuration
-  React 18 Support: Native Next.js APIs are now supported, as well as Suspense
-  `<Image />` AVIF Support: Opt-in for 20% smaller images
-  Bot-aware ISR Fallback: Optimized SEO for web crawlers
- Native ES Modules Support: Aligning with the standardized module system
- URL Imports (alpha): Import packages from any URL, no installs required
- React Server Components (alpha): Try it today, including SSR streaming

Here's the [upgrade guide](https://nextjs.org/docs/upgrading)

### PR DESCRIPTION
**Pull request by DukeManh**
This PR updates deps and removes `babel.config.js` so that `next` opts in using the faster SWC compiler. However, I couldn't get all e2e tests to pass. It looks like the mock function of the `GithubContributorCard` is not working.

close #2401 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by humphd** [comment]
Look like the snapshots need to get regenerated

**Comment by DukeManh** [comment]
> Look like the snapshots need to get regenerated

The snapshot is current, it's the test that failed.

File: src/web/src/components/AboutFooter.test.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { render, cleanup } from '@testing-library/react';
-import AboutFooter from './AboutFooter';
 
 afterEach(cleanup);
 
 jest.mock('./GitHubContributorCard', () => () => <div>GitHubContributorCard</div>);
 
 it('renders correctly', () => {
+  const AboutFooter = require('./AboutFooter').default;
```
Comment (DukeManh) [inline src/web/src/components/AboutFooter.test.tsx:8]: for some reason, the `GitHubContributorCard` mock is not hoisted before the import of `AboutFooter` and returns null. I had to use `require` to make it work.

File: package.json [hunk 2]
```diff
@@ -99,7 +99,6 @@
     "@typescript-eslint/eslint-plugin": "4.17.0",
     "@typescript-eslint/parser": "4.17.0",
     "@vercel/node": "1.9.1",
-    "babel-jest": "27.3.1",
```
Comment (humphd) [inline package.json:102]: Hurray!

File: src/web/jest.config.js [hunk 3]
```diff
@@ -10,11 +15,11 @@ module.exports = {
   collectCoverageFrom: ['src/**/*{ts,tsx}'],
   transform: {
     '^.+\\.(ts)$': 'ts-jest',
-    '^.+\\.(tsx)$': '<rootDir>/node_modules/babel-jest',
+    '^.+\\.(tsx)$': ['babel-jest', { presets: ['next/babel'] }],
```
Comment (humphd) [inline src/web/jest.config.js:18]: So we still have to define `babel-jest` here even though you are removing?  Can you clarify for me so I understand?

File: src/web/src/components/AboutFooter.test.tsx (hunk 1 above)
Comment (humphd) [reply src/web/src/components/AboutFooter.test.tsx:8]: This needs a comment above it explaining why, so that we don't remove it.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Comment by humphd** [comment]
@DukeManh great job.  Rebase and merge this when you have a min.

**Comment by DukeManh** [comment]
@humphd @manekenpix , can you please re-review please.

**Review by manekenpix — DISMISSED** [review]
:+1:

**Review by humphd — APPROVED** [review]
Looks great, ship it.

Question: are there any next.js v12 features we should be adding in follow-up issues/PRs?

**Comment by DukeManh** [comment]
> Question: are there any next.js v12 features we should be adding in follow-up issues/PRs?

I think [React server component](https://nextjs.org/blog/next-12#react-server-components) is interesting. With RSC, JavaScript is computed on the server-side,  generated HTML is progressively streamed to the client with zero JavaScript.

**Comment by DukeManh** [comment]
> I think React server component is interesting.

It requires React 18 which is still beta, and RSC is still alpha.

**Review by humphd — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Pull request merged**